docs/tutorial/connection-monitor.md

```diff
@@ -0,0 +1,519 @@
+# Connection Monitor
+
+Many applications require some sort of online/offline detection mechanism.
+As detailed in the [Online/Offline Event Detection](./online-offline-events.md)
+guide, the chromium browser does provide a mechanism for determining
+connectivity; however, this mechanism is trivial. It only checks for the
+presance of a _network_ connection on the machine and not if it can actually
+connect to a given resource. This example builds a connection monitoring
+service that pings a given resource over a given interval. Depending on the
+result of that request, the service changes its state from **CONNECTED** to
+**DISCONNECTED** and shares this information with the rest of application. In
+addition, the same service is wired up to a common authentication library,
+[@azure/msal-node](https://www.npmjs.com/package/@azure/msal-node) to
+demonstrate how the service can be expanded to support a production-ready
+authentication flow.
+
+This example was built with the intent to be a starting point for other
+projects. You'll most likely have to changes things to suit your own needs.
+Certain parts of the example were designed for this exact reason. For example,
+the `authenticationOperationService` and the `pingOperationService` functions
+are generalized in a way where you may only need to modify the internal logic
+of the function and the state machine can remain the same (more details later).
+
+Additionally, this example is built on the
+[XState](https://github.com/statelyai/xstate) library. The connection monitor
+service is a state machine and utilizes the XState API heavily. Knowledge of
+state machines is not required to follow along with this example, but is
+encouraged. For more information as well as an useful introduction to state
+machines, read through the [XState documentation](https://xstate.js.org/docs/).
+
+Before getting started with the example, if you intend on running this locally
+with `@azure/msal-node` you'll need to include some environment variables. The
+following section will walk you through everything you need to do.
+
+## Example
+
+Once configured, the application should run entirely from Electron Fiddle.
+Later sections of this page will introduce and explain the different parts of
+the example such as the Ping machine, Connection Monitor machine, the IPC
+communication API, and the HTML rendering.
+
+To run this example **without** setting up @azure/msal-node, jump to the
+[Example without @azure/msal-node](#example-without-azuremsal-node) section.
```

Comment:
I'd go about this by setting it up without the auth module and then add that in rather than the other way around.

Reply:
I agree. I think this would allow me to omit a 'fake auth' implementation too. It could be like:
 
step 1. implement ping
step 2. implement connection monitor
(optional) step 3. implement auth state for connection monitor